jdaviz/app.py

```diff
@@ -2655,6 +2657,32 @@ def _get_data_item_by_id(self, data_id):
         return next((x for x in self.state.data_items
                      if x['id'] == data_id), None)
 
+    def _update_data_in_data_collection(self, msg, data_added):
+        """
+        Update the ``data_in_data_collection`` state dictionary to reflect
+        whether data with a given label is in the internal ``DataCollection``.
+
+        Parameters
+        ----------
+        msg : `~glue.core.message.DataCollectionAddMessage` or
+              `~glue.core.message.DataCollectionDeleteMessage`
+            The Glue data collection add or delete message containing
+            information about the new or removed data.
+        data_added : bool
+            Whether data was added or removed from the ``DataCollection``.
+        """
+        # No need to initialize a dictionary of labels in data_in_data_collection
+        # since vue evaluates an empty key as undefined => False
+        update_dict = self.data_in_data_collection.copy()
```

Comment:
instead of doing this, you should be able to update with `self.data_in_data_collection = {**self.data_in_data_collection, 'changed_key': changed_value}`